pkg/controller/policy_controller.go

```diff
@@ -134,8 +135,14 @@ func (pc *PolicyController) processAllocDecision() {
 
 		for _, p := range ps.Running {
 			if len(p.NodeName) == 0 {
-				// TODO(k82cn): it's better to use /eviction instead of delete to avoid race-condition.
-				if err := pc.kubeclient.CoreV1().Pods(p.Namespace).Delete(p.Name, &metav1.DeleteOptions{}); err != nil {
+				// use /eviction instead of delete to avoid race-condition.
+				evict := &policyv1.Eviction{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      p.Name,
+						Namespace: p.Namespace,
+					},
+				}
+				if err := pc.kubeclient.CoreV1().Pods(p.Namespace).Evict(evict); err != nil {
```

Comment:
Does `Evict` work as we expected? If not, I'd like to keep this TODO and raise a requirement to upstream.

Reply:
`Evict` will terminate the pod directly. It will not change the pod to pending status for next scheduling.

Reply:
If Pod was deleted (similar behaviour to Delete), I'd like to keep TODO here.